cmake/modules/BuildSPDK.cmake

```diff
@@ -24,7 +24,7 @@ macro(build_spdk)
       INTERFACE_INCLUDE_DIRECTORIES "${source_dir}/include")
     list(APPEND SPDK_LIBRARIES spdk::${c})
   endforeach()
-  list(APPEND iface_libs ${DPDK_LIBRARIES} rt)
+  list(APPEND iface_libs ${DPDK_LIBRARIES} uuid rt)
```

Comment:
please note, uuid is not a dependency of `spdk::env_dpdk`, it is a dependency of `libspdk_lvol`. we need to 

1. detect libuuid-dev or using `find_package(libuuid REQUIRED)` unless we have a way to disable it, but AFAICT, it's not optional in SPDK. please refer to 3170f16e864e257416bccdd73777bff775767a8f
1. add libuuid-dev{,el} back to `ceph.spec.in` and `debian/control`.
1. add `lvol` to the component list at https://github.com/ceph/ceph/pull/19096/files#diff-05c598c78a24f8547e136669dd4429ccR19.
1. and add `INTERFACE_LINK_LIBRARIES` for `spdk::lvol`

Reply:
and the new DPDK version needs libnuma-dev. we need ensure it's either disabled or our building script can fulfill the requirement.

Reply:
@optimistyzy @tchaikov 
Dependend on the calls it doew not require uuid to be included because routines are in libc.
Or it needs to be added as dependancy in make-deps.sh as well.
But then than would need to be the e2fsprogs-libuuid package

Reply:
@wjwithagen i don't know what "calls" or "routines" or "libc" you were looking at. but take `uuid_compare()` as an example, it is not part of POSIX, C standard,or any extension. see https://linux.die.net/man/3/uuid_compare.

> But then than would need to be the e2fsprogs-libuuid package

please feel free to do it. but please note that SPDK is disabled on FreeBSD.

Reply:
> The `libuuid` was removed from CMake files during one of the library sweep cleanups you did.

actually i was referencing that very commit in my reply of https://github.com/ceph/ceph/pull/19096#discussion_r152490524

> This is what the FreeBSD manpage gives for `man uuid`:

TIL !

Reply:
@wjwithagen SPDK also supports FreeBSD

Reply:
@optimistyzy 
I that a question or a remark? 
Yes, SPDK is supported on FreeBSD.
But the way you require libuuid will break on FreeBSD, because the `uuid_*` routines are already resolved by `libc`. So please only add `uuid` only for other platforms.